### PR TITLE
node: Add publish-docs job

### PR DESCRIPTION
**Pull request by busunkim96**
See https://github.com/googleapis/nodejs-pubsub/pull/548

### CODE REVIEW

File: synthtool/gcp/templates/node_library/.kokoro/publish-docs.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# Publish documentation with docuploader.
+python3 -m pip install --user gcp-docuploader
+
+VERSION=$(npm view {{ metadata['name'] }} version)
+
+# TODO(busunkim): Add product-page and issue-tracker
+python3 -m docuploader create-metadata \
+			--name {{ metadata['product'] }} \
+			--version ${VERSION}\
+			--language node \
```
Comment (busunkim96) [inline synthtool/gcp/templates/node_library/.kokoro/publish-docs.sh]: `node` or `nodejs`?

File: synthtool/gcp/templates/node_library/.kokoro/publish-docs.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+npm install
+
+npm run docs
+
+# Publish documentation with docuploader.
+python3 -m pip install --user gcp-docuploader
+
+VERSION=$(npm view {{ metadata['name'] }} version)
+
+# TODO(busunkim): Add product-page and issue-tracker
```
Comment (busunkim96) [inline synthtool/gcp/templates/node_library/.kokoro/publish-docs.sh]: Is there a list somewhere of products and corresponding issue trackers? https://cloud.google.com/support/docs/issue-trackers?

File: synthtool/languages/node.py [hunk 3]
```diff
@@ -35,6 +35,7 @@ def read_metadata():
 
         repo = git.parse_repo_url(data["repository"])
 
+        data["product"] = data["name"].rsplit("/")[-1]
```
Comment (busunkim96) [inline synthtool/languages/node.py]: Most of the time this works
nodejs-pubsub and https://cloud.google.com/pubsub/

But sometimes the names are a little different... :woman_shrugging::
@google-cloud/debug-agent vs. https://cloud.google.com/debugger/

File: synthtool/gcp/templates/node_library/.kokoro/publish-docs.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# Copyright 2018 Google LLC
```
Comment (JustinBeckwith) [inline synthtool/gcp/templates/node_library/.kokoro/publish-docs.sh]: Heh, nit: 2019

File: synthtool/gcp/templates/node_library/.kokoro/publish-docs.sh (hunk 1 above)
Comment (JustinBeckwith) [reply synthtool/gcp/templates/node_library/.kokoro/publish-docs.sh]: prefer nodejs pls :)

File: synthtool/languages/node.py (hunk 3 above)
Comment (JustinBeckwith) [reply synthtool/languages/node.py]: As discussed - let's create a 2 pager that makes a proposal for a metadata file that sits in the root of the repo.  Keeping in mind, this will apply on a per-directory basis for mono-repo languages like python.

File: noxfile.py [hunk 5]
```diff
@@ -42,3 +44,22 @@ def generate_protos(session):
...
+    session.run(
+        'sphinx-build',
+        '-W',  # warnings as errors
+        '-T',  # show full traceback on exception
+        '-N',  # no colors
+        '-b', 'html',
+        '-d', os.path.join('docs', '_build', 'doctrees', ''),
+        os.path.join('docs', ''),
+        os.path.join('docs', '_build', 'html', ''),
+    )
```
Comment (JustinBeckwith) [inline noxfile.py]: Please add a new line so the angry icon doesn't yell at me

File: synthtool/gcp/templates/node_library/.kokoro/continuous/node10/publish-docs.cfg [hunk 6]
```diff
@@ -0,0 +1,4 @@
+env_vars: {
```
Comment (busunkim96) [inline synthtool/gcp/templates/node_library/.kokoro/continuous/node10/publish-docs.cfg]: Didn't mean to add these kokoro jobs to continuous/ and presubmit/, they'll  be removed in another commit.

**Comment by bcoe** [comment]
I apologize, this PR wasn't on my radar when I started work on:

https://github.com/googleapis/synthtool/pull/249

We're consolidating effort now.
